hangman.py

```diff
@@ -0,0 +1,348 @@
+        if len(letters_guessed) == 1:
+            if letters_guessed in letters:
+                if warnings > 0:
+                    warnings -= 1
+                else:
+                    warnings = "no"
+                    text = " so you lose one guess"
+                    tries -= 1
+                print("Oops! You've already guessed that letter. You have", warnings, "warnings left" + text + ":",
+                      check)
+            elif letters_guessed not in string.ascii_lowercase:
+                if warnings > 0:
+                    warnings -= 1
+                else:
+                    warnings = "no"
+                    text = " so you lose one guess"
+                    tries -= 1
+                print("Oops! That is not a valid letter. You have", warnings, "warnings left:", check)
+            elif letters_guessed in secret_word_list:
+                letters_guessed_list.append(letters_guessed)
+                check = get_guessed_word(secret_word, letters_guessed_list)
+                print("Good guess:", check)
+            elif letters_guessed not in secret_word_list:
+                if letters_guessed in vowels_list:
+                    tries -= 2
+                else:
+                    tries -= 1
+                check = get_guessed_word(secret_word, letters_guessed_list)
+                print("Oops! That letter is not in my word:", check)
+            letters.append(letters_guessed)
+        elif len(letters_guessed) < 1:
+            if warnings > 0:
+                warnings -= 1
+            else:
+                warnings = "no"
+                text = " so you lose one guess"
+                tries -= 1
+            print("Oops! Don't leave a blank line. You have", warnings, "warnings left:", check)
+        else:
+            if warnings > 0:
+                warnings -= 1
+            else:
+                warnings = "no"
+                text = " so you lose one guess"
+                tries -= 1
+            print("Oops! Don't enter more than one symbol. You have", warnings, "warnings left:", check)
+        if tries == 0:
+            print("-" * 13)
+            print('Sorry, you ran out of guesses. The word was "' + secret_word + '".')
+            break
+        if secret_word == check:
+            score = tries * len(set(secret_word))
+            print("-" * 13)
+            print("Congratulations, you won! Your total score for this game is:", score)
+            break
+    return
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    my_word = "".join(my_word.split())
```

Comment:
Why do you split the string, and then join it? Why you haven't try to use a `replace` function?